### PR TITLE
Terrance Danckert FSW13 (db-zoos)

### PR DESCRIPTION
**Pull request by Tbarnett20**
Setup Knex

### CODE REVIEW

**Comment by Tbarnett20** [comment]
Spelled realized wrong lol to anyone who sees this...I know

File: index.js [hunk 1]
```diff
@@ -1,14 +1,90 @@
 const express = require('express');
 const helmet = require('helmet');
+const knex = require('knex');
```
Comment (gooseandmegander) [inline index.js]: You are using `knex` and `sqlite3` modules, but they aren't dependencies in your project. How is your code running?

File: index.js [hunk 2]
```diff
@@ -1,14 +1,90 @@
...
 server.use(helmet());
 
-// endpoints here
 
-const port = 3300;
+/*########################################### POST ###########################################*/
+
+/************************ POST Zoo's ************************/
+server.post('/api/zoos', (req, res) => {
+  const zoo = req.body;
```
Comment (gooseandmegander) [inline index.js]: What happens if there is no name in the req.body? Why would you want to check that before sending the req.body to the db? What if I send `{ "name": "" }`? Will that go through and _should_ it?

File: index.js [hunk 3]
```diff
@@ -1,14 +1,90 @@
...
+/*########################################### DELETE ###########################################*/
+
+/************************ DELETE Zoo's ************************/
+server.delete('/api/zoos/:id', (req, res) => {
+  const { id } = req.params;
+   db('zoos')
+    .where({ id })
+    .delete(id)
+    .then(count => {
+      if (!count || count < 1) {
```
Comment (gooseandmegander) [inline index.js]: 👍

File: index.js [hunk 4]
```diff
@@ -1,14 +1,90 @@
...
+    .catch(err => res.status(500).json(err));
+});
+
+/*########################################### PUT ###########################################*/
+
+/************************ PUT Zoo's ************************/
+
+server.put('/api/zoos/:id', (req, res) => {
+  const {id} = req.params;
+  const name = req.body;
```
Comment (gooseandmegander) [inline index.js]: What happens if there is no name in the req.body? Why would you want to check that before sending the req.body to the db? What if I send `{ "name": "" }`? Will that go through and _should_ it?

File: index.js [hunk 5]
```diff
@@ -1,14 +1,90 @@
...
+  db('zoos')
+    .where({id})
+    .update(name)
+    .then( zoos => {
+      res.status(200).json(zoos);
```
Comment (gooseandmegander) [inline index.js]: What if the `id` isn't in the db? How would you handle that? _Hint: you already handled this case in another endpoint._

**Pull request closed without merging**